src/components/IBCDenomInput.tsx

```diff
@@ -0,0 +1,128 @@
+import { Fragment, useMemo, useState } from "react";
+import { Combobox, Transition } from "@headlessui/react";
+import { CheckIcon, ChevronUpDownIcon } from "@heroicons/react/20/solid";
+import { useQuery } from "@tanstack/react-query";
+import { ibcDenomTracesQuery, ibcDenomHashQuery } from "../lib/queries";
+import { useNetwork } from "../hooks/useNetwork";
+import { DenomTrace } from "../types/bank";
+import { selectSinglePathDenomTraces } from "../lib/selectors";
+
+const TraceToHash = ({
+  path,
+  baseDenom,
+}: {
+  path: string;
+  baseDenom: string;
+}) => {
+  const { api } = useNetwork();
+  const ibcHashTrace = useQuery(ibcDenomHashQuery(api, path, baseDenom));
+  if (!ibcHashTrace.data) return null;
+  return (
+    <>
+      {ibcHashTrace.data}
+      <input type="hidden" name="denom" value={ibcHashTrace.data} />
+    </>
+  );
+};
+
+const formatTrace = (trace: DenomTrace): string => {
+  if (!trace) return "";
+  const { base_denom, path } = trace;
+  return `${base_denom} ${path}`;
+};
+
+const IBCDenomInput = () => {
```

Comment:
Should the user be allowed to make a proposal for an IBC denom that isn't on the/a running chain?

I'm doing some testing on a local chain and I have nothing to select.

Reply:
Good idea. See [80e9796](https://github.com/DCFoundation/cosmos-proposal-builder/pull/30/commits/80e97967a40ea6bbc45d2cde88df5ef703c6a8d5)

Reply:
While trying to confirm that this works, I hit a bug that I assume affects the app before this PR:

 - #31

It lets me type in `ibc/blort`, so otherwise, I suppose it's working.